gnd/src/main/java/com/google/android/gnd/workers/TileDownloadWorker.java

```diff
@@ -150,27 +153,23 @@ private Result checkDownload(Tile tile) {
   @NonNull
   @Override
   public Result doWork() {
-    Tile tile = localDataStore.getTile(tileId).blockingGet();
+    ImmutableList<Tile> pendingTiles = localDataStore.getPendingTiles().blockingGet();
 
-    // When there is no tile in the db, the Maybe completes and returns null.
+    // When there are no tiles in the db, the blockingGet returns null.
     // We expect tiles to be added to the DB prior to downloading.
     // If that isn't the case, we fail.
-    if (tile == null) {
+    if (pendingTiles == null) {
```

Comment:
We shouldn't care if there are no pending tiles; if so another instance of the worker may have completed the job for us, and we should just exit since everything is in the proper state.

Reply:
Gotcha, we'll just treat it as a success then eec1754